WordPress/gutenberg
# Site editor: bump the z-index of the rename modal

## What?
This PR bumps the rename modal `z-index` to be one higher than the dropdown popover component, which is [set](https://github.com/WordPress/gutenberg/blob/819172cb3c1200654903eb56f9a694580aba4f39/packages/base-styles/_z-index.scss#L120-L120) to `1000000`.

I noticed this while working on https://github.com/WordPress/gutenberg/pull/54271

## Why?
So that the rename modal sits above the actions dropdown:

<img width="400" alt="Screenshot 2023-09-08 at 2 32 23 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/8695d09c-4699-49eb-928f-0dfcba39ce34">
<img width="481" alt="Screenshot 2023-09-08 at 2 29 15 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/74f728ab-fde7-46e3-a1de-f014823050e2">



## Testing Instructions
Create a custom template and pattern. In the site editor, click on the actions ellipsis in the individual views and rename. Also check the patterns grid. 

To ensure that modal hovers over the open dropdown element, reduce the dimensions of your browser.

The modal and its background should be above the dropdown. 

Thank you!

## Screenshots or screencast 

<img width="660" alt="Screenshot 2023-09-08 at 2 36 14 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/e74da432-9559-4643-a15e-ee5df709c471">
<img width="815" alt="Screenshot 2023-09-08 at 2 35 56 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/348e88f9-691b-4ad1-a014-e853a2a8537a">
